### PR TITLE
[WIP] Update variables data types

### PR DESCRIPTION
**Pull request by raymondtfr**

### CODE REVIEW

File: src/item.cpp [hunk 1]
```diff
@@ -751,7 +751,7 @@ void Item::serializeAttr(PropWriteStream& propWriteStream) const
 
 	if (hasAttribute(ITEM_ATTRIBUTE_DEFENSE)) {
 		propWriteStream.write<uint8_t>(ATTR_DEFENSE);
-		propWriteStream.write<int32_t>(getIntAttr(ITEM_ATTRIBUTE_DEFENSE));
+		propWriteStream.write<uint32_t>(getIntAttr(ITEM_ATTRIBUTE_DEFENSE));
```
Comment (ranisalt) [inline src/item.cpp]: This is backwards-incompatible, do take care. It is better to wait until #2206 for items is ready.

**Comment by ranisalt** [comment]
Keep in mind it is not necessarily advantageous to have unsigned values for non-negative fields e.g. `Position`'s `x`, `y` and `z`. Sometimes you will want to subtract and handle overflows, which are not easily detectable with unsigned values.

File: src/item.cpp (hunk 1 above)
Comment (raymondtfr) [reply src/item.cpp]: Oh okay... so should I put this PR on hold for now?

Comment (ranisalt) [reply src/item.cpp]: You might keep working on it for science, but I would call it waste of time *for now*.

**Comment by DSpeichert** [comment]
While these values may not make sense to be negative, this looks like a change for the sake of change as there is no other explanation attached. Unless the change brings any benefits, it's not worth merging.

**Pull request closed without merging**